src/utils/auth/auth.utils.ts

```diff
@@ -136,7 +139,14 @@ export async function createAuthPopup(
     const hasWallets = activeAddress && wallets.length > 0;
 
     if (!hasWallets) {
-      openOrSelectWelcomePage(true);
+      if (import.meta.env?.VITE_IS_EMBEDDED_APP === "1") {
+        postEmbeddedMessage({
```

Comment:
If the dApp tries to use `window.arweaveWallet` before the user is authenticated or a wallet is loaded and activated (this last part will not be required later, in order to make it load faster), the popup opens automatically.